clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -3382,7 +3382,43 @@ public void testCommittedThrowsTimeoutExceptionForNoResponse(GroupProtocol group
         assertEquals("Timeout of 1000ms expired before the last committed offset for partitions [test-0] could be determined. " +
             "Try tuning default.api.timeout.ms larger to relax the threshold.", timeoutException.getMessage());
     }
-    
+
+    @ParameterizedTest
+    @EnumSource(value = GroupProtocol.class)
+    public void testPreventMultiThread(GroupProtocol groupProtocol) throws InterruptedException {
```

Comment:
Could you please consider writing it by UT. https://github.com/apache/kafka/blob/4a3ab89f95aba294bb536af55548522d946d1ee3/clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java#L2428 is a good example

Reply:
Hi @chia7712 , thanks for your suggestion! Updated it!